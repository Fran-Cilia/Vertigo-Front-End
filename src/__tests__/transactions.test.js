import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import TransactionsLayout from '../Pages/TransactionsPage/TransactionsPage'
import renderer from 'react-test-renderer'
import ToggleGroup from '../Pages/TransactionsPage/TransactionsPage'



describe('RENDER <TransactionLayout />', () =>{
    afterEach(() => {
        cleanup();
    });
    
    it('Renders Transactions Components', () => {
        render(<TransactionsLayout/>)

        const buttonHeader = screen.getByTestId('Header')
        expect(buttonHeader).toBeInTheDocument();
        
        const publicButton = screen.getByTestId('Public')
        expect(publicButton).toBeInTheDocument();
        expect(publicButton).toHaveTextContent('Public')

        const friendsButton = screen.getByTestId('Friends')
        expect(friendsButton).toBeInTheDocument();
        expect(friendsButton).toHaveTextContent('Friends')
    })

    it('Matches Snapshot', () => {
        const tree = renderer.create(<TransactionsLayout/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('<ToggleGroup />', () => {
    afterEach(() => {
        cleanup();
    });

    it('OnClick Public', () => {
        const { queryByText } = render(<ToggleGroup/>);

        const publicButton = queryByText('Public');
        const friendsButton = queryByText('Friends');
        fireEvent.click(publicButton);

        expect(publicButton.className).toBe('Active')
        expect(friendsButton.className).toBe('Inactive')
    })

    it('OnClick Friends', () => {
        const { queryByText } = render(<ToggleGroup/>);

        const publicButton = queryByText('Public');
        const friendsButton = queryByText('Friends');
        fireEvent.click(friendsButton);

        expect(publicButton.className).toBe('Inactive')
        expect(friendsButton.className).toBe('Active')
    })
})