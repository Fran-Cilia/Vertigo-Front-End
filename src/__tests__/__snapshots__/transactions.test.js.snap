// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDER <TransactionLayout /> Matches Snapshot 1`] = `
<div
  className="Header"
  data-testid="Header"
>
  <button
    className="Active"
    data-testid="Public"
    onClick={[Function]}
  >
    Public
  </button>
  <button
    className="Inactive"
    data-testid="Friends"
    onClick={[Function]}
  >
    Friends
  </button>
</div>
`;
